waspc/examples/todoApp/src/apis.ts

```diff
@@ -1,10 +1,9 @@
-import { getFirstProviderUserId } from "wasp/auth";
-import { type MiddlewareConfigFn } from "wasp/server";
-import { type BarBaz, type FooBar, type WebhookCallback } from "wasp/server/api";
+import { type MiddlewareConfigFn } from 'wasp/server'
+import { type BarBaz, type FooBar, type WebhookCallback } from 'wasp/server/api'
 import express from 'express'
 
 export const fooBar: FooBar = (_req, res, context) => {
-  const username = getFirstProviderUserId(context?.user) ?? 'Anonymous'
+  const username = context.user?.getFirstProviderUserId()
```

Comment:
Ok cool this is nicer now!
What does `getFirstProviderUserId` promise? If I have multiple providers, it will return id from one of them. And it will be consistent in the order, right? What if I add a new provider? Maybe this ID will become different, right? That might be ok,I am just curious.

Reply:
It will return the first available provider :D the order might change with if you add a new provider.

I predict that this method will be mostly used by people who only have one provider anyways or just want to display _something_ identifying the user. For proper "production usage" I suspect people will want to put some sort of an "email" or "username" on the `User` object anyways.

Reply:
Yeah, I agree! We should make sure to mention this in the docs of this function, to make it clear how it is intended to be used, and what the pitfalls might be, and why they should therefore potentially put stuff on `User` instead. I don't think this stuff will be clear to a new Wasp user.

Reply:
I mentioned putting stuff on `User` here: https://github.com/wasp-lang/wasp/pull/1915/files#diff-1da5418e699dcbebea8ab516c4e60235efd7191abad17057916ca5f9bba2ddb1R263

But I didn't comment on the pitfalls of the `getFirstProviderUserId` method.